### PR TITLE
chore: add bench for revm

### PR DESCRIPTION
**Pull request by ImJeremyHe**
Which docs this PR relation:

Ref #

Which toolchain this PR adaption:

No Breaking Change

Special notes for your reviewer:

CI Switch
ci-runs-only: [Chaos CI, Cargo Clippy, Coverage Test, E2E Tests, Code Format, Unit Tests, Web3 Compatible Tests]

### CODE REVIEW

**Comment by KaoImin** [comment]
Please fix the CI and change the benchmark transaction count to 10000. Besides, record the benchmark result with the test hardware in annotation.

File: core/executor/benches/bench_vm.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+};
+use std::sync::Arc;
+
+use core_executor::{AxonExecutor, AxonExecutorAdapter, MPTTrie};
+use protocol::{
+    codec::ProtocolCodec,
+    traits::{Executor, Storage},
+    types::{Account, Address, ExecutorContext},
+};
+use revm_adapter::{revm_exec, RevmAdapter};
```
Comment (KaoImin) [inline core/executor/benches/bench_vm.rs]: Notice the order of imports, such as: 
```rust
use std::aaa;

use criterion::bbb;

use protocol::ccc;
use core_executor::ddd;

use crate::mock::eee;
```

File: core/executor/benches/mock.rs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+use core_storage::ImplStorage;
+use protocol::{
+    codec::hex_decode,
+    types::{
+        public_to_address, Account, Address, Bytes, Eip1559Transaction, ExecutorContext, Public,
+        SignedTransaction, TransactionAction, UnsignedTransaction, UnverifiedTransaction, H160,
+        H256, H512, NIL_DATA, RLP_NULL, U256,
+    },
+};
+use std::sync::Arc;
```
Comment (KaoImin) [inline core/executor/benches/mock.rs]: ditto

File: core/executor/benches/mock.rs [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+lazy_static::lazy_static! {
```
Comment (KaoImin) [inline core/executor/benches/mock.rs:19]:
```suggestion

lazy_static::lazy_static! {
```

File: core/executor/benches/mock.rs [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    result
+}
+
+#[inline]
+fn rand_hash() -> H256 {
+    use rand::random;
+    H256::from_slice(&(0..32).map(|_| random()).collect::<Vec<_>>())
+}
```
Comment (KaoImin) [inline core/executor/benches/mock.rs]: Use `H256::random()` instead.
https://docs.rs/ethereum-types/0.13.1/ethereum_types/struct.H256.html#method.random

File: core/executor/benches/revm_adapter.rs [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+use common_merkle::Merkle;
+use core_executor::{code_address, MPTTrie};
+use evm::ExitSucceed;
+use protocol::codec::ProtocolCodec;
+use protocol::traits::{Context, Storage};
+use protocol::types::{
+    Account, Address, Bytes, ExecResp, ExecutorContext, Hasher, Proposal, SignedTransaction,
+    TransactionAction, TxResp, H160, H256, NIL_DATA, RLP_NULL, U256,
+};
+use revm::{Bytecode, Database, DatabaseCommit};
```
Comment (KaoImin) [inline core/executor/benches/revm_adapter.rs]: ditto

**Review by KaoImin — CHANGES_REQUESTED** [review]

File: core/executor/benches/revm_adapter.rs [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+    let result: &[u8] = bytemuck::cast_slice(u64_slice);
+    result
+}
+
+#[inline]
+fn u8_slice_to_u256(bytes: &[u8]) -> Option<U256> {
+    let u64_slice: &[u64] = bytemuck::cast_slice(bytes);
+    let array: &[u64; 4] = u64_slice.try_into().expect("incorrect length");
+    Some(U256(array.to_owned()))
+}
```
Comment (KaoImin) [inline core/executor/benches/revm_adapter.rs:221]: Why not use `U256::to_le_bytes(&self)` and `U256::from_le_bytes` method.

Comment (ImJeremyHe) [reply core/executor/benches/revm_adapter.rs:221]: I have tried writing such like this:
```rust
    use protocol::types::U256;

    fn u256_to_u8_slice_2(index: U256) -> Vec<u8> {
        let mut bytes = vec![];
        index.to_little_endian(&mut bytes);
        bytes
    }

    fn u8_slice_to_u256_2(bytes: &[u8]) -> U256 {
        U256::from_little_endian(bytes)
    }

    #[test]
    fn convert_test() {
        let num = U256::from(12345678910099u64);
        let slice = u256_to_u8_slice_2(num);
        let actual = u8_slice_to_u256_2(&slice);
        assert_eq!(num, actual);
    }
```

And it went wrong.

Output:
```
running 1 test
thread 'system::tests::convert_test' panicked at 'assertion failed: 4 * 8 == bytes.len()', /Users/jeremyhe/.cargo/registry/src/github.com-1ecc6299db9ec823/primitive-types-0.11.1/src/lib.rs:38:1
```

Comment (KaoImin) [reply core/executor/benches/revm_adapter.rs:221]: The `to_little_endian` require the given vector with the 32 length. You  can write like this:
```rust
let mut bytes = vec![0u8; 32];
index.to_little_endian(&mut bytes); 
bytes
```

File: core/executor/benches/bench_vm.rs [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+use mock::{
+    init_account, mock_executor_context, mock_transactions, new_rocks_trie_db, new_storage,
+};
+use protocol::{
+    codec::ProtocolCodec,
+    traits::{Executor, Storage},
+    types::{Account, Address, ExecutorContext},
+};
+
+use revm_adapter::{revm_exec, RevmAdapter};
```
Comment (KaoImin) [inline core/executor/benches/bench_vm.rs:18]: Move this import location.

File: core/executor/benches/revm_adapter.rs (hunk 6 above)
Comment (ImJeremyHe) [reply core/executor/benches/revm_adapter.rs:221]: OK.
I wrote a benchmark for these 2 methods:
```rust
use criterion::{criterion_group, criterion_main, Criterion};

use protocol::types::U256;

fn u256_to_u8_slice(index: &U256) -> &[u8] {
    let u64_slice = index.as_ref();
    let result: &[u8] = bytemuck::cast_slice(u64_slice);
    result
}

fn u256_to_u8_slice_2(index: U256) -> Vec<u8> {
    let mut bytes = vec![0u8; 32];
    index.to_little_endian(&mut bytes);
    bytes
}

fn u8_slice_to_u256_2(bytes: &[u8]) -> U256 {
    U256::from_little_endian(bytes)
}

fn u8_slice_to_u256(bytes: &[u8]) -> Option<U256> {
    let u64_slice: &[u64] = bytemuck::cast_slice(bytes);
    let array: &[u64; 4] = u64_slice.try_into().expect("incorrect length");
    Some(U256(array.to_owned()))
}

fn criterion_convert(c: &mut Criterion) {
    let num = U256::from(12345678910099u64);
    c.bench_function("convert1", |b| {
        b.iter(|| {
            let slice = u256_to_u8_slice(&num);
            let actual = u8_slice_to_u256(slice).unwrap();
            assert_eq!(num, actual);
        });
    });
    c.bench_function("convert2", |b| {
        b.iter(|| {
            let slice = u256_to_u8_slice_2(num);
            let actual = u8_slice_to_u256_2(&slice);
            assert_eq!(num, actual);
        });
    });
}

criterion_group!(benches, criterion_convert);
criterion_main!(benches);

```

And the result:
- convert1: 858.84ps
- convert2: 32.415ns

If this benchmark should be put into this PR?

Comment (KaoImin) [reply core/executor/benches/revm_adapter.rs:221]: It is up to you.

**Review by KaoImin — APPROVED** [review]

**Pull request merged**